src/dnstest_reporter.erl

```diff
@@ -1,14 +1,15 @@
 -module(dnstest_reporter).
+-include_lib("kernel/include/logger.hrl").
 
 -export([report/1]).
 
+-spec report([dnstest_harness:return()]) -> ok.
 report(TestResults) ->
-    PassFail = lists:map(
-        fun({Name, Result}) ->
-            [Name, lists:all(fun(R) -> R end, Result)]
-        end,
-        TestResults
-    ),
-    {Pass, Fail} = lists:partition(fun([_, Result]) -> Result end, PassFail),
-    lager:info("~p Passed, ~p Failed", [length(Pass), length(Fail)]),
-    lists:foreach(fun(R) -> lager:info("~p: ~p", R) end, Fail).
+    {Pass, Fail} = lists:partition(fun(#{result := Result}) -> true =:= Result end, TestResults),
+    ?LOG_INFO(#{
+        passed_num => length(Pass),
+        passed => lists:map(fun(#{name := Name}) -> Name end, Pass),
+        failed_num => length(Fail),
+        failed => lists:map(fun(#{name := Name}) -> Name end, Fail)
+    }),
```

Comment:
It has become very hard to read the output now.

Reply:
I could remove the failed/passed elements, it might probably make the output smaller, and also configure the logger to not output timestamps as they're not needed. However one thing I was missing when reading out the output is knowing why a record was failing so hopefully the `lists:foreach(fun(Return) -> ?LOG_INFO(Return) end, Fail).` below can help with that (?)